### PR TITLE
Update test-cases.md

### PR DESCRIPTION
**Pull request by sofisanabria**

### CODE REVIEW

**Review by roxanafalco20 — APPROVED** [review]
Green light on my end :)

File: Test-DigitalBank/test/depositflow.test.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    await HomePage.backtologinpage();
+  });
+
+  dataDepositFlow.forEach((depositData) => {
+    it(`Create a deposit with amount ${depositData.amount}`, async () => {
+        await DepositPage.selectAccountByPosition(depositData.index);
+        await DepositPage.fillAmount(depositData.amount);
+        const expectedBalance = await DepositPage.getBalanceAmount() + parseFloat(depositData.amount);
+        await DepositPage.clickSubmitButton();
+        await expect(AccountView.balanceText).toHaveTextContaining(expectedBalance)
```
Comment (santiago-abstracta) [inline Test-DigitalBank/test/depositflow.test.js:27]: You can also verify if the deposit was actually made by checking the corresponding table.

**Review by santiago-abstracta — CHANGES_REQUESTED** [review]

File: Test-DigitalBank/data/depositflowtest.js [hunk 2]
```diff
@@ -0,0 +1,4 @@
+module.exports = [
```
[the remark this answers is not in the document]
Comment (santiago-abstracta) [reply Test-DigitalBank/data/depositflowtest.js]: Done.

File: Test-DigitalBank/data/depositflowtest.js [hunk 3]
```diff
@@ -0,0 +1,4 @@
...
+    {index: 1, amount: "500"},
+    {index: 2, amount: "1000"}
```
[the remark this answers is not in the document]
Comment (santiago-abstracta) [reply Test-DigitalBank/data/depositflowtest.js]: Done.

File: Test-DigitalBank/pages/deposit.page.js [hunk 4]
```diff
@@ -28,10 +24,6 @@ class DepositPage extends BasePage {
     get errorMessage(){
         return $("span.badge-danger")
     }
-
-    async goToDepositPage(){
-        await this.clickElement(this.depositPage);
-    }
     
     async findDepositMessage(){
```
[the remark this answers is not in the document]
Comment (santiago-abstracta) [reply Test-DigitalBank/pages/deposit.page.js]: Done.

File: Test-DigitalBank/pages/deposit.page.js [hunk 5]
```diff
@@ -53,5 +45,11 @@ class DepositPage extends BasePage {
         await this.clickElement(this.submitButton);
     }
 
+    async getBalanceAmount(){
```
[the remark this answers is not in the document]
Comment (santiago-abstracta) [reply Test-DigitalBank/pages/deposit.page.js]: We still need to have that conversation!

[the remark this answers is not in the document]
Comment (santiago-abstracta) [reply Test-DigitalBank/pages/deposit.page.js]: Done.

File: Test-DigitalBank/pages/home.page.js [hunk 6]
```diff
@@ -13,6 +13,9 @@ class HomePage extends BasePage {
   get withdrawPageMenuItem(){
```
[the remark this answers is not in the document]
Comment (santiago-abstracta) [reply Test-DigitalBank/pages/home.page.js]: Done.

File: Test-DigitalBank/pages/home.page.js [hunk 7]
```diff
@@ -13,6 +13,9 @@ class HomePage extends BasePage {
...
     return $("#withdraw-menu-item")
   }
+  get depositPage(){
```
[the remark this answers is not in the document]
Comment (santiago-abstracta) [reply Test-DigitalBank/pages/home.page.js]: Done.

File: Test-DigitalBank/pages/home.page.js [hunk 8]
```diff
@@ -13,6 +13,9 @@ class HomePage extends BasePage {
...
+    return $("#deposit-menu-item");
+  }
 
   async backtologinpage() {
```
[the remark this answers is not in the document]
Comment (santiago-abstracta) [reply Test-DigitalBank/pages/home.page.js]: I understand now! I believe this method should be called `logOut()`.

File: Test-DigitalBank/wdio.conf.js [hunk 9]
```diff
@@ -1,3 +1,6 @@
+import {addStep} from '@wdio/allure-reporter'
```
Comment (santiago-abstracta) [inline Test-DigitalBank/wdio.conf.js:1]: Good! It'd be great to see steps in your tests :)

File: Test-DigitalBank/test/withdraw.test.js [hunk 10]
```diff
@@ -15,7 +15,7 @@ describe('Withdraw', () => {
     });
     afterEach(async () => {
         await browser.url(' ');
```
Comment (santiago-abstracta) [inline Test-DigitalBank/test/withdraw.test.js:17]: You can call the `logOut()` method from any page as it depends on an element that's always present, so this line isn't necessary.

File: Test-DigitalBank/test/login.test.js [hunk 11]
```diff
@@ -12,7 +12,7 @@ describe("LogIn test", () => {
     await LoginPage.clickLogInButton();
     const welcometext = await HomePage.getWelcomeText();
     await expect(welcometext).toEqual("Welcome Josh");
```
Comment (santiago-abstracta) [inline Test-DigitalBank/test/login.test.js:14]: This also works: `await expect(await HomePage.getWelcomeText()).toEqual("Welcome Josh")`.

File: Test-DigitalBank/test/depositflow.test.js [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+import dataDepositFlow from "../data/depositflowtest";
+import AccountView from "../views/account.view"
+
+describe("Deposit Flow", () => {
+  before(async () => {
+    await HomePage.open("");
+  });
+  beforeEach(async () => {
+    await LoginPage.fillLogInForm("jsmith@demo.io", "Demo123!");
+    await LoginPage.clickLogInButton();
```
[the remark this answers is not in the document]
Comment (santiago-abstracta) [reply Test-DigitalBank/test/depositflow.test.js]: Done.

File: Test-DigitalBank/test/depositflow.test.js [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+import HomePage from "../pages/home.page";
+import DepositPage from "../pages/deposit.page";
+import dataDepositFlow from "../data/depositflowtest";
+import AccountView from "../views/account.view"
+
+describe("Deposit Flow", () => {
+  before(async () => {
+    await HomePage.open("");
+  });
+  beforeEach(async () => {
```
[the remark this answers is not in the document]
Comment (santiago-abstracta) [reply Test-DigitalBank/test/depositflow.test.js:12]: Thanks for reaching out to me!

File: Test-DigitalBank/views/account.view.js [hunk 14]
```diff
@@ -0,0 +1,13 @@
+import BasePage from "../pages/base.page";
+
+class AccountView extends BasePage {
+    
+    get pageTitle(){
+        return $("#page-title")
+    }
+    get balanceText(){
+        return $(".card-body").$("div:nth-child(8)")
+    }
```
Comment (santiago-abstracta) [inline Test-DigitalBank/views/account.view.js:10]: What happens if there's more than one card in the account view?

File: Test-DigitalBank/test/depositflow.test.js (hunk 1 above)
Comment (santiago-abstracta) [reply Test-DigitalBank/test/depositflow.test.js:21]: I believe it's better to check if the first row corresponds to the last deposit made.

File: Test-DigitalBank/test/depositflow.test.js [hunk 15]
```diff
@@ -0,0 +1,24 @@
...
+    await HomePage.goToDepositPage();
+  });
+
+  deposits.forEach((deposit) => {
+    it(`should deposit ${deposit.amount} into ${deposit.account}`, async () => {
+        await DepositPage.deposit(deposit.account,deposit.amount);
+        const expectedBalance = await DepositPage.getBalance() + parseFloat(deposit.amount);
+        await DepositPage.clickSubmitButton();
```
Comment (santiago-abstracta) [inline Test-DigitalBank/test/depositflow.test.js:20]: Clicking this button should be part of the `deposit()` method.

File: Test-DigitalBank/test/deposit.test.js [hunk 16]
```diff
@@ -10,14 +10,13 @@ describe('Deposits', () => {
...
-        await LoginPage.clickLogInButton();
-        await DepositPage.goToDepositPage();
+        await LoginPage.login();
+        await HomePage.goToDepositPage();
       });
     
     afterEach(async () => {
         await browser.url(' ');
-        await HomePage.backtologinpage();
+        await HomePage.backToLoginPage();
```
Comment (santiago-abstracta) [inline Test-DigitalBank/test/deposit.test.js:19]: `logOut()`

File: Test-DigitalBank/pages/login.page.js [hunk 17]
```diff
@@ -24,5 +24,10 @@ class LoginPage {
   async getErrorMessage() {
     return await this.errorMessage.getValue();
   }
+
+  async login(){
+    await this.fillLogInForm("jsmith@demo.io", "Demo123!");
```
Comment (santiago-abstracta) [inline Test-DigitalBank/pages/login.page.js:29]: Let's talk about this through Google Meet.

**Review by santiago-abstracta — CHANGES_REQUESTED** [review]

File: Test-DigitalBank/test/deposit.test.js [hunk 18]
```diff
@@ -12,7 +12,7 @@ describe('Deposits', () => {
     beforeEach(async () => {
         await LoginPage.fillLogInForm("jsmith@demo.io", "Demo123!");
         await LoginPage.clickLogInButton();
```
[the remark this answers is not in the document]
Comment (santiago-abstracta) [reply Test-DigitalBank/test/deposit.test.js]: Done.

File: Test-DigitalBank/test/depositflow.test.js (hunk 13 above)
[the remark this answers is not in the document]
Comment (santiago-abstracta) [reply Test-DigitalBank/test/depositflow.test.js:12]: Done.

File: Test-DigitalBank/test/depositflow.test.js [hunk 19]
```diff
@@ -0,0 +1,30 @@
...
+    await LoginPage.fillLogInForm("jsmith@demo.io", "Demo123!");
+    await LoginPage.clickLogInButton();
+    await HomePage.goToDepositPage();
+  });
+  afterEach(async () => {
+    await HomePage.open("");
+    await HomePage.backtologinpage();
+  });
+
+  dataDepositFlow.forEach((depositData) => {
```
[the remark this answers is not in the document]
Comment (santiago-abstracta) [reply Test-DigitalBank/test/depositflow.test.js]: Done.

File: Test-DigitalBank/test/withdraw.test.js [hunk 20]
```diff
@@ -15,7 +15,7 @@ describe('Withdraw', () => {
...
-        await HomePage.backtologinpage();
+        await HomePage.backToLoginPage();
     })
 
     it('Sould show the withdraws page', async () => {
```
Comment (santiago-abstracta) [inline Test-DigitalBank/test/withdraw.test.js:21]: should*

**Review by santiago-abstracta — CHANGES_REQUESTED** [review]

File: Test-DigitalBank/test/depositflow.test.js [hunk 21]
```diff
@@ -0,0 +1,30 @@
...
+    it(`Create a deposit with amount ${depositData.amount}`, async () => {
```
[the remark this answers is not in the document]
Comment (santiago-abstracta) [reply Test-DigitalBank/test/depositflow.test.js]: Done.

File: Test-DigitalBank/test/depositflow.test.js [hunk 22]
```diff
@@ -0,0 +1,30 @@
...
+        await DepositPage.selectAccountByPosition(depositData.index);
+        await DepositPage.fillAmount(depositData.amount);
```
[the remark this answers is not in the document]
Comment (santiago-abstracta) [reply Test-DigitalBank/test/depositflow.test.js]: Done.

File: Test-DigitalBank/test/depositflow.test.js [hunk 23]
```diff
@@ -0,0 +1,25 @@
...
+    await HomePage.open("login");
+    await LoginPage.login();
+  });
+  beforeEach(async () => {
+    await HomePage.goToDepositPage();
+  });
+
+  deposits.forEach((deposit) => {
+    it(`should deposit ${deposit.amount} into ${deposit.account}`, async () => {
+      addStep("Create Deposit")
```
Comment (santiago-abstracta) [inline Test-DigitalBank/test/depositflow.test.js:18]: I'd like to see more steps in the next Nano!

**Review by santiago-abstracta — APPROVED** [review]

**Pull request merged**